crates/biome_css_analyze/src/lint/nursery/no_duplicate_selectors.rs

```diff
@@ -0,0 +1,327 @@
+            if let Some(parent_rule) = AnyCssAtRule::cast_ref(&parent_rule) {
+                let mut hasher = DefaultHasher::new();
+                parent_rule.range().hash(&mut hasher);
+                // Each @rule is unique scope
+                // Use a hash to create the comparable scope
+                parent_selectors.push(hasher.finish().to_string());
+            }
+            if let Some(parent_rule) = AnyCssRule::cast_ref(&parent_rule) {
+                match parent_rule {
+                    AnyCssRule::CssNestedQualifiedRule(parent_rule) => {
+                        for selector in parent_rule.prelude() {
+                            if let Ok(selector) = selector {
+                                parent_selectors.push(selector.text());
+                            }
+                        }
+                    }
+                    AnyCssRule::CssQualifiedRule(parent_rule) => {
+                        for selector in parent_rule.prelude() {
+                            if let Ok(selector) = selector {
+                                parent_selectors.push(selector.text());
+                            }
+                        }
+                    }
+                    _ => {
+                        // Bogus rules are not handled
+                        // AtRule is handled by AnyCssAtRule above
+                    }
+                }
+            }
+
+            let resolved_selectors: Vec<String> =
+                parent_selectors
+                    .iter()
+                    .fold(vec![], |result: Vec<String>, parent_selector| {
+                        if selector.contains("&") {
+                            let resolved_parent_selectors = resolve_nested_selectors(
+                                parent_selector.to_string(),
+                                parent_rule.clone(),
+                            );
+                            let resolved = resolved_parent_selectors
+                                .into_iter()
+                                .map(|newly_resolved| return selector.replace("&", &newly_resolved))
+                                .collect();
+                            return [result, resolved].concat();
+                        } else {
+                            let combined_selectors = parent_selector.to_owned() + " " + &selector;
+                            let resolved =
+                                resolve_nested_selectors(combined_selectors, parent_rule.clone());
+                            return [result, resolved].concat();
+                        }
+                    });
+            if resolved_selectors.len() > 0 {
+                return resolved_selectors;
+            }
+            return vec![selector];
+        }
+    }
+}
+
+fn normalize_complex_selector(selector: AnyCssSelector) -> String {
+    let mut selector_text = String::new();
+
+    if let Some(complex_selector) = CssComplexSelector::cast_ref(&selector.clone().into_syntax()) {
```

Comment:
```suggestion
    if let AnyCssSelector::CssComplexSelector(complex_selector) = selector {
```

No need cloning and `into_syntax`